core/state/state_object.go

```diff
@@ -410,18 +432,24 @@ func (s *stateObject) setBalance(amount *big.Int) {
 }
 
 func (s *stateObject) deepCopy(db *StateDB) *stateObject {
-	stateObject := newObject(db, s.address, s.data)
+	obj := &stateObject{
+		db:       db,
+		address:  s.address,
+		addrHash: s.addrHash,
+		origin:   s.origin,
+		data:     s.data,
```

Comment:
I think this only deep copies the pointer to s.data, I'm not sure if this is what you want

Reply:
s.data is a struct, it's ok to directly copy it.

Re s.origin, seems we will never change the content of origin via pointer, it's also OK to copy the pointer.